launcher/ui/pages/modplatform/ModModel.cpp

```diff
@@ -192,13 +192,14 @@ void ListModel::searchRequestFinished(QJsonDocument& doc)
 
 void ListModel::searchRequestFailed(QString reason)
 {
-    if (jobPtr->first()->m_reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt() == 409) {
+    if (!jobPtr->first()->m_reply) {
+        // Network error
+        QMessageBox::critical(nullptr, tr("Error"), tr("A network error occurred. Could not load mods."));
+    } else if (jobPtr->first()->m_reply && jobPtr->first()->m_reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt() == 409) {
         // 409 Gone, notify user to update
         QMessageBox::critical(nullptr, tr("Error"),
                               //: %1 refers to the launcher itself
                               QString("%1 %2").arg(m_parent->displayName()).arg(tr("API version too old!\nPlease update %1!").arg(BuildConfig.LAUNCHER_NAME)));
-        // self-destruct
-        (dynamic_cast<ModDownloadDialog*>((dynamic_cast<ModPage*>(parent()))->parentWidget()))->reject();
```

Comment:
Why remove it? It's really awkward having an empty dialog after the error. Maybe you can create a `downloadDialog()` in ModPage, and use that to reject it in case of a fail in the network request.

Reply:
Main reason to remove it was, that it caused a segfault when I added the same to my new code. Apart from that, it's very possible that if one Mod API fails (CurseForge in the future probably) that the other might still work

Reply:
Hmm yes, ideally we would check the other API if the first one failed, and switch to it if it was available, otherwise we would close the dialog, since there's no point, imo. But that's not a hotfix, so it's fine as it is!